src/lib.rs

```diff
@@ -358,66 +359,72 @@ impl<'a> FMatcher<'a> {
-                        }
-                        if ptnm.start() > text.len() || ptn[..ptnm.start()] != text[..ptnm.start()]
-                        {
-                            return false;
-                        }
-                        ptn = &ptn[ptnm.end()..];
-                        text = &text[ptnm.start()..];
-                        if let Some(textm) = text_re.find(text) {
-                            if self.options.distinct_name_matching {
-                                for (x, y) in names.iter().chain(new_names.iter()) {
-                                    if x != &ptnm.as_str() && y == &textm.as_str() {
-                                        return false;
-                                    }
-                                }
+            if self.options.name_matchers.is_empty() {
+                if eww {
+                    text.starts_with(&ptn[..ptn.len() - WILDCARD.len()])
+                } else {
+                    ptn == text
+                }
+            } else {
+                let mut new_names = HashMap::new();
+                loop {
+                    let mut matched = false;
+                    for (ref ptn_re, ref text_re) in &self.options.name_matchers {
+                        if let Some(ptnm) = ptn_re.find(ptn) {
+                            matched = true;
+                            if ptnm.start() == ptnm.end() {
+                                panic!("Name pattern matched the empty string.");
                             }
-                            if textm.start() == textm.end() {
-                                panic!("Text pattern matched the empty string.");
+                            if ptnm.start() > text.len()
+                                || ptn[..ptnm.start()] != text[..ptnm.start()]
+                            {
+                                return false;
                             }
-                            match names.entry(ptnm.as_str()) {
-                                Entry::Occupied(e) => {
-                                    if e.get() != &textm.as_str() {
-                                        return false;
+                            ptn = &ptn[ptnm.end()..];
+                            text = &text[ptnm.start()..];
+                            if let Some(textm) = text_re.find(text) {
+                                if self.options.distinct_name_matching {
+                                    for (x, y) in names.iter().chain(new_names.iter()) {
+                                        if x != &ptnm.as_str() && y == &textm.as_str() {
+                                            return false;
+                                        }
                                     }
                                 }
-                                Entry::Vacant(_) => match new_names.entry(ptnm.as_str()) {
+                                if textm.start() == textm.end() {
```

Comment:
You could shorten this to:
```
assert_eq!(textm.start(),  textm.end(), "Text pattern matched the empty string.");
```

Reply:
I think this should stay a `panic` because it's the result of an error on the user's part. To me an `assert` should reflect an error on my/our part when developing `fm`. There is an element of angels-on-a-pin in this, though, I admit.

Reply:
So you prefer the semantic meaning of "panic" to "assert". I ask because in Rust an `assert!` is just a dressed up `panic!`.

Reply:
Yes and yes.

Reply:
OK